crates/uv-extract/src/sync.rs

```diff
@@ -1,21 +1,37 @@
+use std::hash::BuildHasherDefault;
 use std::path::{Path, PathBuf};
-use std::sync::Mutex;
 
+use dashmap::mapref::entry::Entry;
 use rayon::prelude::*;
-use rustc_hash::FxHashSet;
+use rustc_hash::FxHasher;
 use zip::ZipArchive;
 
 use crate::vendor::{CloneableSeekableReader, HasLength};
 use crate::Error;
 
+type FxDashMap<K, V> = dashmap::DashMap<K, V, BuildHasherDefault<FxHasher>>;
```

Comment:
Since we're using this as a set, I'd find it clearer to define this as a `FxDashSet<V>` and fix the value type to `()`.

Reply:
I think I'd also add a comment here saying why we aren't using a `DashSet` directly. I assume it's because a `DashSet` has no API for holding the lock in the case of a non-existent key.